### ISSUE TITLE
No errors handling during connection

### ISSUE THREAD

**Issue by msztorc** [voryx/Thruway#147]
Hi everyone,

I have a problem with a php client from your examples.

When I try to connect to server with disabled ws routing then client is frozen and debug shows info: "[Thruway\Transport\PawlTransportProvider 1156] Starting Transport". It does not show any error, just hangs. The same situation is when connection is closed (disconnected). The connection will not be restored.

**Comment by davidwdan** [comment]
@msztorc what class are you using, `Connection` or `Client` and what version of Thruway?

**Comment by msztorc** [comment]
Hi @davidwdan 

I'm using `Client` class.  
Version: "voryx/thruway": "^0.3.9"

**Comment by davidwdan** [comment]
@mbonneau can you take a look at this?  I suspect it might have something to do with Pawl.

**Comment by mbonneau** [comment]
Hi @msztorc ,

I am trying to reproduce this issue. Are you saying that if you connect to regular web server (a web server that cannot handle websockets) it just hangs? Or are you connecting to something else?

I have tried it against an apache server and it seems to detect the failure.

Any more information on how to reproduce this issue would be great.

**Comment by msztorc** [comment]
Hi @mbonneau 

Sorry for my imprecision. You have right, it's hard to reproduce this issue on a unix based systems. I'm using Cygwin as a client and Debian as a websocket server. 
If I try connect from client running on a cygwin to remote websocket server when ws server is up then everything works fine, but if remote server has gone away, then client hangs.

**Comment by msztorc** [comment]
Hi @mbonneau 

It seems the problem is only with php for Windows. First, I was thinking about Cygwin, because I forgot about existed installation of php on my Windows. When I uninstalled php from Windows and after installed php through Cygwin installer everything works fine.

I was tested this script on other Windows 7 machines with cleanly installed php x64 versions 5.5, 5.6 and 5.7 and script everytime hangs after connection lost. Maybe problem is with configuration in php.ini or openssl extension? 

In conclusion:
Cygwin with enabled php: works fine
Php for Windows: hangs after connection lost (not trying to reconnect)

**Comment by mbonneau** [comment]
@msztorc ,

I have not had a chance to test this in a non-cygwin installation yet. I am trying to track down a machine to do that on and will get back to you.

**Comment by raditv** [comment]
it's happen also on laravel homestead. when I stop crossbar router, client hangs

**Comment by Wapweb** [comment]
I have the same issue: clients are frozen when i stopped the server (php script with Router).
Even if server running again after failed, clients still frozen. but i guess clients must be try to reconnect.
My environment:
windows 7 x64
php 5.5
voryx/thruway 0.4.2

**Comment by RafaelKa** [comment]
Hey folks, sorry for asking but why are you using Windows for services like WAMP? Why you can't use proper system? Please meet Vagrant or much better Docker!

**Comment by davidwdan** [comment]
Our goal is for Thruway to run on any system that supports the minimum requirements, **which includes windows**. 

We haven't been able to figure out what is causing this issue, so we'll leave this open until we collect enough information to fix it.

@RafaelKa a "proper" system should be defined by the requirements. I think it's fair to say that we don't know what @Wapweb's requirements are in this case, so second guessing his decision to use windows, is not helping him solve his problem.  There are perfectly acceptable use cases for running Thruway on windows.

**Comment by RafaelKa** [comment]
@davidwdan i think supporting Windows is a waste of time.
Check https://www.docker.com/microsoft . No needs to support native execution on Windows, if microsoft itself set on Docker ;)

Postgraduate for windows users https://docs.docker.com/machine/drivers/hyper-v/

**Comment by davidwdan** [comment]
@RafaelKa Like I said, the requirements should dictate what technology is used.  This is a longstanding practice.  All of the docker containers in the world won't help if you **need** to run php on windows. 

We do not consider helping people who want to use our project a "waste of time" and yes that does includes you too;-)

**Comment by ledazinha** [comment]
I've been using Linux for a few days now, but before that I was developing applications using WAMP for almost a year in a Windows environment, without any problem. I never saw the problem mentioned by the OP, while connecting either to a remote or a local websocket server that went down.

I would like to know what version of PHP the OP was utilizing. Was it VC9/VC11/VC14? What is x86 ou x64? My setup used to be: PHP 5.6.3 VC11 x64 .

**Comment by msztorc** [comment]
@RafaelKa  I'm using Thruway on windows for some specific reasons. 
I need processing some files through software which running only for Windows and I have to send task to Windows machine that fast as possible. WAMP protocol is perfect for this. I decide to using Thruway, because I'm using this library on my other servers with Linux and works perfect and it's very stable. 

I "solved" this problem running ws-router script using php from Cygwin instead php from windows console.

**Comment by Wapweb** [comment]
I found solution for this issue!
I've installed libevent library from [Pecl repository](https://pecl.php.net/package/libevent/0.0.5/windows) and the problem is fixed!
And other features like [this](https://github.com/voryx/Thruway/issues/42#issuecomment-86805502) now works correctly

So i guess this bug related to default event loop api

**Comment by msztorc** [comment]
@Wapweb You're right, it works now.

**Comment by megafinance** [comment]
I have same problem like @msztorc. Mac OS 10.12.3 and PHP 7.1.1 (cli).

**Comment by szymonpt** [comment]
I have same problem like @msztorc but on OS X 10.12.3 and PHP 7.1.1 (cli) from brew. Connection is just resetted (TCP RST received) and no event fired.
